internal/partinfo/fat32.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2023 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package partinfo
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+)
+
+const (
+	fat32Sig          = "FAT32"   // FAT32 filesystem type signature
+	emptyLabel        = "NO NAME" // Default empty volume label
+	mbrSig     uint16 = 0xAA55    // MBR signature
+)
+
+type fat32Partition struct {
+	f          *os.File
+	superblock struct {
+		_      [71]byte
+		Label  [11]byte // Volume label
+		FSType [8]byte  // FAT filesystem type
+		_      [420]byte
+		MBRSig uint16 // MBR signature
+	}
+}
+
+func (p fat32Partition) Path() string {
+	return p.f.Name()
+}
+
+func (p fat32Partition) FSType() string {
+	return "vfat"
+}
+
+func (p fat32Partition) Label() string {
+	if s := strings.TrimSpace(string(p.superblock.Label[:])); s != emptyLabel {
+		return s
+	} else {
+		return ""
+	}
```

Comment:
Nit: early returns and avoiding the `else` / putting the happy path up an indentation level is customary in Go. So I'd suggest:

```suggestion
	label := strings.TrimSpace(string(p.superblock.Label[:]))
	if s == emptyLabel {
		return ""
	}
	return label
```